### ISSUE TITLE
support plotly orca

### ISSUE THREAD

**Issue by iynehz** [pablrod/p5-Chart-Plotly#8]
Latest R plotly 4.8.0 supports orca. It allows saving as static images. https://github.com/plotly/orca

**Comment by pablrod** [comment]
I'm not sure but I think that you want something like this:

```perl
my $plot = ...;
system('orca', 'graph', to_json($plot), '-o', 'example-export.png');
```
Or maybe this:

```perl
my $plot = ...;
$plot->export_image(filename => 'example-export.png');
```

**Comment by iynehz** [comment]
Basically I would like to see an API in Chart::Plotly, rather than calling system() by people themselves. 

There seems to have multiple possible ways of implementation, see 
https://github.com/plotly/plotly.py/issues/1105
https://github.com/ropensci/plotly/blob/master/R/orca.R

We might need to study it and see which way could be best for Chart::Plotly. 

Actually I am currently working on something that could potentially help to bring Perl to the table of data analysis. I will contact you offline once I have something demo-able.

**Comment by pablrod** [comment]
I've done a first implementation using directly the orca command tool.

The intended usage can be seen in https://github.com/pablrod/p5-Chart-Plotly/blob/d1612d90f689137cac03f409bdb5a303383b052e/examples/orca.pl 

```perl
Chart::Plotly::Image::Orca::orca(plot => $plot, file => "TestOrca.png");
```

This command takes around 350 ms on my laptop. For several plots this can be fairly slow so I think that the orca server support is a must.

Also I was using [Selenium::Remote::Driver](https://metacpan.org/pod/Selenium::Remote::Driver) (see [tool/generate_images_from_examples.pl](https://github.com/pablrod/p5-Chart-Plotly/blob/master/tool/generate_images_from_examples.pl)) maybe I can support both behind a "Image Exporting" interface.

I'm not sure if this is the right API, maybe a method in `Chart::Plotly::Plot` is better but please feel free to comment! Thank you!

**Comment by iynehz** [comment]
@pablrod thanks!

I just created a PR #11 to make some minor changes. 

As for the performance, I think 350ms is fast! On my Ubuntu 18.04 Virtualbox guest it's about 4 seconds for that simple example, and it's 27 seconds for my example using the "diamonds" data set. Maybe it relates to the fact that I disabled the guest's 3D acceleration as using with plotly orca it caused Xorg to crash... I will need to find that out later. 

IMHO it's fine for now to have it in such a functional api for studying the matter in this branch. We probably can add an api in Chart::Plotly::Plot once we think it's good enough.

**Comment by pablrod** [comment]
PR #11 merged! Thanks!

> IMHO it's fine for now to have it in such a functional api for studying the matter in this branch. We probably can add an api in Chart::Plotly::Plot once we think it's good enough.

I agree with you that this api is enough for now. I was going to make a new release with this but I'll wait if you prefer.

**Comment by iynehz** [comment]
Yes please merge it and have it in your next release to CPAN. btw, I am not 100% sure but I guess you might want to update the pod of Chart::Plotly::Image::Orca a little bit, as I see you use dzil+podweaver, while you seemed to had created this module by copying from a dzil+podweaver generated module.

**Comment by pablrod** [comment]
Thanks! I'll make the release between tonight and tomorrow!

Yes you're right I'm messing a little bit with dzil+podweaver but the mistake is in other modules also (see also Chart::Plotly::Plot). It's my fault. I'll fix Chart::Plotly::Image::Orca now and the rest of modules will follow (but maybe not for the release)

**Comment by pablrod** [comment]
With pull request #18 merged and released in 0.024, the orca support is good enough for me! So I'm going to close this issue.

What do you think @stphnlyd ? Do you miss something?

**Comment by iynehz** [comment]
@pablrod yes this can be closed. Thanks!
